src/App.tsx

```diff
@@ -1,60 +1,149 @@
 import './App.scss';
+import React, { useState } from 'react';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { User } from './types/user';
+import { Todo } from './types/todo';
+import { TodoList } from './components/TodoList';
+
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
 
 export const App = () => {
+  const [todosForRender, setTodos] = useState(todos);
+
+  const [title, setTitle] = useState('');
+  const [UserID, addUserId] = useState(0);
+  const [titleErr, setTitleError] = useState(false);
+  const [userErr, setUserIdError] = useState(false);
+
+  const refreshForm = () => {
+    setTitle('');
+    addUserId(0);
+  };
+
+  const inputValue = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setTitleError(false);
+  };
+
+  function generateId() {
+    const max = todosForRender.reduce((largest, x) => {
+      return largest > x ? largest : x;
```

Comment:
What is largest? Are you sure that you are comparing todos id?

Reply:
I changed generateID function, but i still dont`t pass one test. In consolen the keys look right 
<img width="184" alt="image" src="https://user-images.githubusercontent.com/106486742/200375436-6c259466-45e2-45e3-b46f-e3b5c2fa7e19.png">